### PR TITLE
Feature/convention

### PR DESCRIPTION
**Pull request by juliusb0205**
@ming-ancino 

Implement bootstrap. Follow conventions

### CODE REVIEW

**Comment by juliusb0205** [comment]
@ming-ancino 

Removed public from repo. Added public in .gitignore

File: app/views/layouts/_header.haml [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+	.container
+		#navbar.navbar-collapse.collapse
+			- if logged_out?
+				.navbar-form.navbar-left
+					= link_to :root , :class => 'btn btn-warning' do
+						%i.glyphicon.glyphicon-home
+							Training Home
+				= form_tag :login , :method => :post, :class => 'navbar-form navbar-right' do
+					.form-group
+						= text_field_tag :email, "", :class => 'form-control', :placeholder => 'Email'
```
Comment (ming-ancino) [inline app/views/layouts/_header.haml]: Use single quotes unless interpolation is needed

File: app/views/layouts/_header.haml [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+					.form-group
+						= password_field_tag :password, "", :class => 'form-control', :placeholder => 'Password'
```
Comment (ming-ancino) [inline app/views/layouts/_header.haml]: Quotes.

File: app/views/questions/_answer_partial.haml [hunk 3]
```diff
@@ -1,17 +1,28 @@
...
-		%li
-			= answer.sum_votes
-			= answer.content
-			= render :partial => 'vote', :locals => { :object => answer }
-			%ul
-				- answer.comments.latest.each do |comment|
-					%li
-						= comment.content
+	- if logged_in?
+		= form_for [@question,@answer] do |f|
```
Comment (ming-ancino) [inline app/views/questions/_answer_partial.haml]: Fix array spacing

File: app/views/questions/_answer_partial.haml [hunk 4]
```diff
@@ -1,17 +1,28 @@
...
+	%table.table
+		%tr
+			%td.vote_cell
+				.vote
+					= render :partial => 'votes/vote', :locals => {:object => answer}
+			%td.content_cell
+				= answer.content
+				%br
+				.pull-right.answer_comment_by
+					Answered by #{link_to answer.user.name, answer.user} on #{answer.created_at.strftime("%m/%d/%Y at %I:%M%p")}
```
Comment (ming-ancino) [inline app/views/questions/_answer_partial.haml]: parenthesis

File: app/views/questions/_answer_partial.haml [hunk 5]
```diff
@@ -1,17 +1,28 @@
...
+				= render :partial => 'comments/comment', :locals => {:object => answer}
+				- if logged_in?
+					%a.custom_button{:data => {:toggle => 'collapse', :target => "#comment#{answer.id}"}}
+						Add Comment
+					.form-group.collapse{ :id => "comment#{answer.id}"}
+						= form_for [answer,@comment] do |f|
```
Comment (ming-ancino) [inline app/views/questions/_answer_partial.haml]: space

File: app/views/questions/_form.haml [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
   %br
-  = f.label :content
-  %br
-  = f.text_area :content
-  %br
-  = f.label :tags
-  %br
-  = f.text_area :tag_list
+  .form-group
+    = f.label :tags, "Tags ( separate multiple tags by comma )"
```
Comment (ming-ancino) [inline app/views/questions/_form.haml]: Quotes

File: app/views/questions/_question_partial.haml [hunk 7]
```diff
@@ -1,25 +1,13 @@
...
 - if logged_in?
-	= render :partial => 'shared/error', :locals => { :object => @comment }
-	= form_for [@question,@comment] do |f|
-		= f.label :comment
-		= f.text_area :content
-		= f.submit 'Submit Comment'
+	%a.custom_button{:data => {:toggle => 'collapse', :target => '#demo'}}
+		Add Comment
+	.form-group.collapse#demo
+		= form_for [@question,@comment] do |f|
```
Comment (ming-ancino) [inline app/views/questions/_question_partial.haml]: space

File: app/views/questions/show.haml [hunk 8]
```diff
@@ -1,17 +1,26 @@
...
-  =@question.sum_votes
-  =@question.title
+%table.table
+  %tr
+    %td.vote_cell
+      .vote-ajax
+        = render :partial => 'votes/vote', :locals => {:object => @question}
+    %td.content_cell
+      %h3.title
+        =@question.title
```
Comment (ming-ancino) [inline app/views/questions/show.haml]: space

File: app/views/questions/show.haml [hunk 9]
```diff
@@ -1,17 +1,26 @@
...
+            %h6.inline
+              = link_to 'Edit', [:edit, @question]
+            %h6.inline
+              = link_to 'Delete', @question, :method => :delete, :data => {:confirm => 'Are you sure you want to delete this question?'}
+      = @question.content
+
 
 %h4
+  .btn.btn-default.disable
+    Tags:
```
Comment (ming-ancino) [inline app/views/questions/show.haml]: space

File: app/views/questions/vote.js.erb [hunk 10]
```diff
@@ -0,0 +1 @@
+$("#vote-ajax").html("<%= j ( render :partial => 'questions/actual_count', :locals => {:object => @vote.voteable}) %>");
```
Comment (ming-ancino) [inline app/views/questions/vote.js.erb]: quotes

File: app/views/shared/_error.haml [hunk 11]
```diff
@@ -4,10 +4,9 @@
 -########
 
 - if object.errors.any?
-  %div.error_explanation
+  .error_explanation
```
Comment (ming-ancino) [inline app/views/shared/_error.haml]: fix quotes etc

File: app/views/users/new.haml [hunk 12]
```diff
@@ -1,24 +1,16 @@
...
-		= f.label :password
-		%br
-		= f.password_field :password
-		%br
-		= f.label :password_confirmation
-		%br
-		= f.password_field :password_confirmation
-		%br
-		= f.submit "Register"
+= form_for @user, :html => { :class => 'form-signin' } do |f|
```
Comment (ming-ancino) [inline app/views/users/new.haml]: space

File: app/views/users/new.haml [hunk 13]
```diff
@@ -1,24 +1,16 @@
...
+	= f.text_field :name, :class => 'form-control', :placeholder => 'Name'
+	= f.text_field :email, :class => 'form-control', :placeholder => 'Email'
+	= f.password_field :password, :class => 'form-control', :placeholder => 'Password'
+	= f.password_field :password_confirmation, :class => 'form-control', :placeholder => 'Password Confirmation'
+	%br
+	.form-inline
+		= f.submit "Register", :class => 'btn btn-success pull-right'
```
Comment (ming-ancino) [inline app/views/users/new.haml]: quotes

File: app/views/votes/_vote.haml [hunk 14]
```diff
@@ -0,0 +1,30 @@
+-#########
+-#locals
+-# object
+-########
+
+%div
+  - if logged_in?
+    - vote = object.votes.vote_from_user(current_user)
```
Comment (ming-ancino) [inline app/views/votes/_vote.haml]: parenthesis

File: app/views/votes/_vote.haml [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    -if vote.blank?
```
Comment (ming-ancino) [inline app/views/votes/_vote.haml]: space

File: app/views/votes/_vote.haml [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+      %div
+        = link_to [object, Vote.new, :vote => {:direction => 1}], :method => :create, :class => 'btn.btn-default' do
```
Comment (ming-ancino) [inline app/views/votes/_vote.haml]: method post?

File: app/views/votes/vote.js.erb [hunk 17]
```diff
@@ -0,0 +1 @@
+$(".vote-ajax").html("<%= j render 'votes/vote', :locals => {:object => @vote.voteable } %>");
```
Comment (ming-ancino) [inline app/views/votes/vote.js.erb]: quotes

File: app/models/question.rb [hunk 18]
```diff
@@ -15,20 +15,33 @@ class Question < ActiveRecord::Base
...
 	end
 
 	def tag_list
 		self.tags.map(&:name).join ','
 	end
 
 	def tag_list= value
-		tags = value.split ','
+		self.tags.delete_all
+		tags = value.strip.split(',').map(&:strip)
```
Comment (ming-ancino) [inline app/models/question.rb]: parenthesis

File: app/models/question.rb [hunk 19]
```diff
@@ -15,20 +15,33 @@ class Question < ActiveRecord::Base
...
 		tags.each do |tag|
-			self.tags << Tag.find_or_create_by(:name => tag)
+			tag_check = Tag.find_or_create_by(:name => tag.downcase)
```
Comment (ming-ancino) [inline app/models/question.rb]: parenthesis

File: app/views/comments/_comment.haml [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+
+%table.table
+  %tbody
+    - object.comments.latest.each do |comment|
+      %tr
+        %td
+          = comment.content
+          %br
+          %div.pull-right.answer_comment_by
+            Commented #{link_to object.user.name, object.user} on #{object.created_at.strftime("%m/%d/%Y at %I:%M%p")}
```
Comment (ming-ancino) [inline app/views/comments/_comment.haml]: parenthesis quotes

File: app/views/questions/_answer_partial.haml (hunk 4 above)
Comment (ming-ancino) [inline app/views/questions/_answer_partial.haml]: quotes parenthesis

**Comment by juliusb0205** [comment]
@ming-ancino 

Corrections updated.

File: app/views/comments/_comment.haml [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+-########
+
+%table.table
+  %tbody
+    - object.comments.latest.each do |comment|
+      %tr
+        %td
+          = comment.content
+          %br
+          %div.pull-right.answer_comment_by
```
Comment (ming-ancino) [inline app/views/comments/_comment.haml]: div not needed

**Pull request merged**